clients/twitter-integration/src/main/java/com/github/ymikevich/twitter/integration/services/DefaultTwitterService.java

```diff
@@ -18,14 +21,23 @@
 
     private final TweetSearchEngine tweetSearchEngine;
     private final TweetProducer tweetProducer;
+    private final UserClient userClient;
 
     @Override
     public List<Tweet> findAndProduceTweetsByUsername(final String username) {
-
         var tweets = tweetSearchEngine.findRecentTweetsByUsername(username);
         log.trace("Sending tweets via rabbitMQ");
         tweetProducer.produce(tweets);
 
         return tweets;
     }
+
+    @Override
+    public void sync(final Long userId) {
+        var userAccounts = ofNullable(userClient.getAccountsByUserId(userId));
+        userAccounts.ifPresent(accountResponses -> accountResponses
+                .stream()
```

Comment:
it's a bad practice to call ForEach in a stream without any intermediate operations

Reply:
or map the getNickname in stream and finish stream with `findAndProduceTweetsByUsername` method reference